### ISSUE TITLE
Timeline or timespecification for scenes/chapter

### ISSUE THREAD

**Issue by ScullyBlue** [olivierkes/manuskript#118]
Its the last thing, I cant migrate to manuskript. The time specifications from scenes or the duration of actions.

For planing its helpful, to have an overview over the time that has passed - since the last scene or between two specific scenes. So its easier to see, that - for example - since the beginning of story ten hours has passed, so its evening. Or its only two hours ago and there is time for more actions or I can jump direct to evening, knowing, that there is a break that must be briged (or I leave it advisedly unmentioned).
Also for parallel Plots it would be a nice feature.

By the way (minor idea): For Sci-Fi-Storys it would be very nice, if each scene can not only have a date and a duration, but also the timeunits can be defined by author (because each planet has its own rotation, so the day can be longer or shorter than on earth).

**Comment by src-r-r** [comment]
I like that idea, too. I'm working on an implementation.

> By the way (minor idea): For Sci-Fi-Storys it would be very nice, if each scene can not only have a date and a duration, but also the timeunits can be defined by author (because each planet has its own rotation, so the day can be longer or shorter than on earth).

And I don't think this is a minor idea. Besides creating different time units Python's date library falls incredibly short for writing fiction (okay, it's really a C issue).
What if someone wants to write a novel that takes place in BC? Or 9 million years in the future? 

Specifying custom time units would work around the issue by providing granularity. Computers store time as an int representing seconds since 1970-ish (if my memory serves me correctly). Chances are, as a writer, you don't care about seconds in a scene, and probably don't care about minutes, either. This opens up the scope of dates that can be represented.

**tl;dr: Yes, good idea.**

**Comment by ScullyBlue** [comment]
I don't thougt about fictions so far in future oder past, but yes, you're right! This would be nice too.

**Comment by jonas-printzen** [comment]
From my perspective this is a more general thing, representing sequence and/or time in a more generic form, binding it to places and cultures, will provide invaluable support for storytelling, especially plot-style stories, RPG campaigns and such. The notion to develop custom time is a very godd way to go and cupod clearly be extended beyond the simplest of concepts. C-time is clearly to rigid in any case.

**Comment by siliconserf** [comment]
Plot entries include metadata references to dramatic beats, etc. If another entry column was supported for text, an author could easily provide time references. Not an in-depth or scaled time line implementation, but would be sufficient for most and go a long way to helping track events with time.

**Comment by Eusorph** [comment]
First of all, the Storyline is very useful and works pretty well. Now, though, I'd love to see an actual Timeline incorporated in the Storyline window. That would be a killer feature, because it would render the use of an external app like Aeon Timeline obsolete. Since you already have a very nice and neat tracking system for all the various items in the story, I think (hope is more the word) that it shouldn't be too difficult to add timestamps as tracking items too, and a Timeline in the same category as Plots and Characters -- although it should work in reverse of course, because timestamps would be put in the Reference window of scenes and not in the Timeline window, unless you want to make it work as another Outline window, I guess, which could work too. That way, we would have an easy and effective Timeline incorporated into Manuskript. Also, I do think the whole Storyline should be able to go full screen or detach to be maximized better.

**Comment by TheJackiMonster** [comment]
I think to get a timeline working with custom units etc. it's most convenient to use a format like this for example: "2001-12-03 13:51:03"
So you could use string comparison and alphabetic order to sort them at least without a representation via integer values.
